src/Core/Components/NavMenu/FluentNavMenu.razor

```diff
@@ -4,15 +4,13 @@
 {
     <FluentPageScript Src="./_content/Microsoft.FluentUI.AspNetCore.Components/Components/NavMenu/FluentNavMenu.razor.js"></FluentPageScript>
 }
-@if (@Collapsible)
-{
-    <FluentKeyCode Anchor="@Id" OnKeyDown="@HandleExpandCollapseKeyDownAsync" />
-}
 <div id="@Id" @ref="@Element" @attributes="AdditionalAttributes" class="@ClassValue" style="@StyleValue" aria-label="@(Collapsible ? null : @Title)" role="menu" aria-expanded="@Expanded">
     <CascadingValue Value="this" IsFixed="true">
         @if (Collapsible)
         {
-            <div aria-label="@Title"
+            <FluentKeyCode Anchor="expander" OnKeyDown="@HandleExpandCollapseKeyDownAsync" />
+            <div id="@EXPAND_COLLAPSE_BUTTON_ID"
```

Comment:
The div ´id` must be unique: you have to use `@Id`, for example adding a private property `ExpandCollapseButtonId => $"{Id}-expander"`